rasa/core/policies/ted_policy.py

```diff
@@ -989,6 +1042,44 @@ def _encode_real_features_per_attribute(
 
             if attribute == TEXT:
                 text_transformer_output = attribute_features
+                text_sequence_lengths = sequence_lengths
+
+                if self.use_only_last_dialogue_turn:
+                    # get the location of all last dialogue inputs
+                    dialogue_lengths = tf.cast(
+                        tf_batch_data[DIALOGUE][LENGTH][0], dtype=tf.int32
+                    )
+                    # TODO precompute dialogue_indices after creation of tf_batch_data
+                    dialogue_indices = (
+                        tf.map_fn(
+                            tf.range,
+                            dialogue_lengths,
+                            fn_output_signature=tf.RaggedTensorSpec(
+                                shape=[None], dtype=tf.int32
+                            ),
+                        )
+                    ).values
+                    last_dialogue_mask = tf.math.logical_not(
+                        tf.cast(
+                            tf.concat(
+                                [dialogue_indices, tf.zeros((1,), dtype=tf.int32)],
+                                axis=0,
+                            )[1:],
+                            dtype=tf.bool,
+                        )
+                    )
```

Comment:
Took me some time to follow the logic, but very smart 💯

Reply:
it's the zeros 😅, it's always a `0`😂

Reply:
which reminds me, I need to add a comment